resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/constants.scala

```diff
@@ -92,25 +91,59 @@ package object constants {
-  private[spark] val INIT_CONTAINER_TRUSTSTORE_SECRET_KEY = "trustStore"
-  private[spark] val INIT_CONTAINER_DOWNLOAD_JARS_SECRET_PATH =
-    s"$INIT_CONTAINER_SECRETS_VOLUME_MOUNT_PATH/$INIT_CONTAINER_DOWNLOAD_JARS_SECRET_KEY"
-  private[spark] val INIT_CONTAINER_DOWNLOAD_FILES_SECRET_PATH =
-    s"$INIT_CONTAINER_SECRETS_VOLUME_MOUNT_PATH/$INIT_CONTAINER_DOWNLOAD_FILES_SECRET_KEY"
-  private[spark] val INIT_CONTAINER_TRUSTSTORE_PATH =
-    s"$INIT_CONTAINER_SECRETS_VOLUME_MOUNT_PATH/$INIT_CONTAINER_TRUSTSTORE_SECRET_KEY"
-  private[spark] val INIT_CONTAINER_DOWNLOAD_CREDENTIALS_PATH =
-    "/mnt/secrets/kubernetes-credentials"
-  private[spark] val INIT_CONTAINER_CONFIG_MAP_KEY = "init-driver"
-  private[spark] val INIT_CONTAINER_PROPERTIES_FILE_VOLUME = "init-container-properties"
-  private[spark] val INIT_CONTAINER_PROPERTIES_FILE_MOUNT_PATH = "/etc/spark-init/"
-  private[spark] val INIT_CONTAINER_PROPERTIES_FILE_NAME = "init-driver.properties"
-  private[spark] val INIT_CONTAINER_PROPERTIES_FILE_PATH =
-    s"$INIT_CONTAINER_PROPERTIES_FILE_MOUNT_PATH/$INIT_CONTAINER_PROPERTIES_FILE_NAME"
-  private[spark] val DOWNLOAD_JARS_VOLUME_NAME = "download-jars"
-  private[spark] val DOWNLOAD_FILES_VOLUME_NAME = "download-files"
+
+  // Init container for downloading submitted files from the staging server.
+  private[spark] val INIT_CONTAINER_SUBMITTED_FILES_CONTAINER_NAME =
+    "spark-driver-download-submitted-files"
+  private[spark] val INIT_CONTAINER_SUBMITTED_FILES_SECRETS_VOLUME_NAME =
+    "resource-staging-server-secret"
+  private[spark] val INIT_CONTAINER_SUBMITTED_FILES_SECRETS_VOLUME_MOUNT_PATH =
+    "/mnt/secrets/spark-init"
+  private[spark] val INIT_CONTAINER_SUBMITTED_FILES_DOWNLOAD_JARS_SECRET_KEY =
+    "downloadSubmittedJarsSecret"
+  private[spark] val INIT_CONTAINER_SUBMITTED_FILES_DOWNLOAD_FILES_SECRET_KEY =
+    "downloadSubmittedFilesSecret"
+  private[spark] val INIT_CONTAINER_SUBMITTED_FILES_TRUSTSTORE_SECRET_KEY = "trustStore"
+  private[spark] val INIT_CONTAINER_SUBMITTED_FILES_DOWNLOAD_JARS_SECRET_PATH =
+    s"$INIT_CONTAINER_SUBMITTED_FILES_SECRETS_VOLUME_MOUNT_PATH/" +
+      s"$INIT_CONTAINER_SUBMITTED_FILES_DOWNLOAD_JARS_SECRET_KEY"
+  private[spark] val INIT_CONTAINER_SUBMITTED_FILES_DOWNLOAD_FILES_SECRET_PATH =
+    s"$INIT_CONTAINER_SUBMITTED_FILES_SECRETS_VOLUME_MOUNT_PATH/" +
+      s"$INIT_CONTAINER_SUBMITTED_FILES_DOWNLOAD_FILES_SECRET_KEY"
+  private[spark] val INIT_CONTAINER_SUBMITTED_FILES_TRUSTSTORE_PATH =
+    s"$INIT_CONTAINER_SUBMITTED_FILES_SECRETS_VOLUME_MOUNT_PATH/" +
+      s"$INIT_CONTAINER_SUBMITTED_FILES_TRUSTSTORE_SECRET_KEY"
+  private[spark] val INIT_CONTAINER_SUBMITTED_FILES_CONFIG_MAP_KEY =
+    "download-submitted-files"
+  private[spark] val INIT_CONTAINER_SUBMITTED_FILES_PROPERTIES_FILE_VOLUME =
+    "download-submitted-files-properties"
+  private[spark] val INIT_CONTAINER_SUBMITTED_FILES_PROPERTIES_FILE_MOUNT_PATH =
```

Comment:
We could download all of the jars from all locations (resource staging server and remote) as well as the files, but I'm concerned about file name conflicts and how to deduplicate those. Perhaps we should explicitly forbid that multiple URIs end with the same file name. @aash for thoughts.